### ISSUE TITLE
Xbox Series X|S Controller Not Detected In Game With Proton 9 And (Some) Native Games

### ISSUE THREAD

**Issue by GildedRoach** [ValveSoftware/Proton#8326]
This issue happens with every game I try, switching to any Proton 8 build at all fixes it,nor did these issues start until Proton 9. Steam detects my controller, other programs detect it, it is _only_ in games. My controller firmware is up to date, I use it wired, and I have tried with and without Xone (patched and unpatched versions for the latest kernels) installed. Protons I have tried are vanilla Proton 9, experimental, and every Proton-GE 9 build.

UPDATE: Maybe related? Some Unity games will not launch under Proton 9, a loading menu of sorts will pop up and then it will crash. House Flipper is on and off an example of this, TCG Card Shop Simulator also does this. Trying to see if I can produce reliable logs.

Currently running on Ultramarine 41 (basically Fedora at the moment).

Small portion of the games I can reproduce this in:

- Slime Rancher (forced in Proton 9, haven't tried native)
- Yooka-Laylee (native or in Proton 9)
- House Flipper
- Borderlands
- Subnautica
- Goat Simulator 3

**Comment by orbea** [comment]
Just a guess, but have you tried turning steam input on or off in the game settings?

**Comment by GildedRoach** [comment]
> Just a guess, but have you tried turning steam input on or off in the game settings?

Ah yes, I forgot to mention, I did do this. It makes the controller work, technically, but then the cursor won't "disappear"/stop inputting (biggest issue) and there are no glyphs at all.

**Comment by SAJewers** [comment]
I have a similar issue, but with an 8bitDo Ultimate C Controller.

EDIT: Using Fedora 41

**Comment by GildedRoach** [comment]
> I have a similar issue, but with an 8bitDo Ultimate C Controller.
> 
> EDIT: Using Fedora 41

Is it also under the same circumstances (i.e., _only_ Proton 9 builds? Ruled out Steam Input issues?)?

**Comment by SAJewers** [comment]
Yep, Proton 9, Proton Experimental, Proton-GE, umu, anything not Valve's Proton 8 (or lower).

Steam Input setting doesn't seem to change anything either

**Comment by andrewbaker-uk** [comment]
Out of interest does doing this fix the issue?

https://github.com/ValveSoftware/Proton/issues/8261#issuecomment-2505870413

**Comment by GildedRoach** [comment]
> Out of interest does doing this fix the issue?
> 
> [#8261 (comment)](https://github.com/ValveSoftware/Proton/issues/8261#issuecomment-2505870413)

Enabling Steam Input in general does make the controller input, however the cursor will not disappear or stop inputting (for example, if the cursor is over an interactable object in a game, it will highlight it) and there are no glyphs.

**Comment by shadow9owo** [comment]
same issue Ubuntu 22.04 Need for speed Heat proton 9

**Comment by shadow9owo** [comment]
i found a way to get my controller to work https://github.com/atar-axis/xpadneo?tab=readme-ov-file

just gonna put it here for others

**Comment by GildedRoach** [comment]
> i found a way to get my controller to work https://github.com/atar-axis/xpadneo?tab=readme-ov-file
> 
> just gonna put it here for others

XPadNeo is for wireless. I have issues using the controller *wired.*

**Comment by shadow9owo** [comment]
> > i found a way to get my controller to work https://github.com/atar-axis/xpadneo?tab=readme-ov-file
> > just gonna put it here for others
> 
> XPadNeo is for wireless. I have issues using the controller _wired._

wel yea thats just a fancy bluetooth driver i dont know about any specific xbox controller drivers though

lemme know if youl find a driver for wired xbox controllers actually i would love to read the source for it

**Comment by GildedRoach** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/8326#issuecomment-2981691702

Depending on the controller, they don't need one.

But I mentioned Xone in my original post (and I use the maintained fork). However, it does not fix my issue.
